examples/system/ulp_riscv/main/ulp/main.c

```diff
@@ -33,10 +33,9 @@ int main (void)
         if(gpio_level != gpio_level_previous) {
             gpio_level_previous = gpio_level;
             ulp_riscv_wakeup_main_processor();
-            break;
```

Comment:
@felmue , the only problem by doing this is actually the ULP will neve enter in its low power mode. The current example only illustrates how the flow works by wakeup and then put all the systems in deep sleep.

I suggest you place a `ulp_riscv_shutdown()` somewhere inside the loop every time the processing ends.